File: module_test/src/main/java/com/gas/test/ui/activity/lifecycle/FirstActivity.java
package com.gas.test.ui.activity.lifecycle;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;

import com.gas.test.R;

import timber.log.Timber;

public class FirstActivity extends AppCompatActivity {


    private static final String TAG=FirstActivity.class.getSimpleName();




    @Override
    protected void onCreate(Bundle savedInstanceState) {

        super.onCreate(savedInstanceState);
        Timber.tag(TAG).e("OnCreate");
        setContentView(R.layout.test_activity_main);
        findViewById(R.id.tv_to_second).setOnClickListener(v ->
                startActivity(new Intent(FirstActivity.this,SecondActivity.class)));
    }

    @Override
    protected void onRestart() {

        super.onRestart();
        Timber.tag(TAG).e("onRestart");
    }

    @Override
    protected void onStart() {

        super.onStart();
        Timber.tag(TAG).e("onStart");
    }

    @Override
    protected void onResume() {

        super.onResume();
        Timber.tag(TAG).e("onResume");
    }

    @Override
    protected void onPause() {
        Timber.tag(TAG).e("onPause");
        super.onPause();
    }

    @Override
    protected void onStop() {
        Timber.tag(TAG).e("onStop");
        super.onStop();
    }

    @Override
    protected void onDestroy() {
        Timber.tag(TAG).e("onDestroy");
        super.onDestroy();
    }
}

File: module_test/src/main/java/com/gas/test/learn/lesson001_lifecycle/FirstActivity.kt
package com.gas.test.learn.lesson001_lifecycle

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import com.gas.test.R
import com.lib.commonsdk.extension.app.debug

class FirstActivity : AppCompatActivity() {


    override fun onNewIntent(intent: Intent?) {
        super.onNewIntent(intent)
        debug("lifecycle","FirstActivity_onNewIntent")
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_first_lifecycle)
        findViewById<View>(R.id.button6).setOnClickListener {
            startActivity(Intent(this, SecondActivity::class.java))
        }
    }
    override fun onRestart() {
        super.onRestart()
        debug("lifecycle","FirstActivity_onRestart")
    }

    override fun onStart() {
        super.onStart()
        debug("lifecycle","FirstActivity_onStart")
    }

    override fun onResume() {
        super.onResume()
        debug("lifecycle","FirstActivity_onResume")
    }


    override fun onPause() {
        super.onPause()
        debug("lifecycle","FirstActivity_onPause")
    }

    override fun onStop() {
        debug("lifecycle","FirstActivity_onStop")
        super.onStop()
    }

    override fun onDestroy() {
        debug("lifecycle","FirstActivity_onDestroy")
        super.onDestroy()
    }
}